non-doc pieces for deployment

kilobyte opened the pull request:
After these commits, a working build of mkdb can be done by untarring and `make`,

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)
<!-- Reviewable:end -->

kilobyte committed:
Build a tarball with memkind included.

bratpiorka commented on src/Makefile:
```diff
@@ -251,7 +244,7 @@ ifeq ($(MALLOC),memkind)
 	DEPENDENCY_TARGETS+= memkind
 	FINAL_CFLAGS+= -DUSE_MEMKIND -I../deps/memkind/include
-	FINAL_LIBS+= ../deps/memkind/.libs/libmemkind.a $(DAXCTL_LIB) -lnuma -ldl
+	FINAL_LIBS+= ../deps/memkind/.libs/libmemkind.a $(DAXCTL_LIB) -lnuma -ldl -hwloc
```
hwloc is not needed in TMDB

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @bratpiorka, @maciekpac, and @PatKamin)

___
*[`src/Makefile` line 247 at r1](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0ZpTU5PyRG1XDlUMfm-r1-247:-N0a5zF2BadDu_W0KovC:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/44e3d0049b9fb5568410e3c477e27d42e09d7625/src/Makefile#L247)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

hwloc is not needed in TMDB
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

PatKamin reviewed (changes requested):
> Reviewed 2 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 3 unresolved discussions (waiting on @bratpiorka, @kilobyte, and @maciekpac)

___
*[`Makefile` line 13 at r3](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aHwHM-KcJd1sYD_z4:-N0aHwHM-KcJd1sYD_z5:b-7itshp) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/7f8714ff9ee1611cbc7b5dab287ee1cf4b3c19b5/Makefile#L13)):*
> ```Makefile
> .PHONY: install
> 
> TARBALL=mtdb-$(shell git describe)
> ```

tmdb?
___
*[`src/Makefile` line 52 at r2](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aHZ2hD-fnWb0qUgUA:-N0aHZ2hD-fnWb0qUgUB:b5xhk5p) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/0e8e96c1981a7e015941b0308afbdee162767ecf/src/Makefile#L52)):*
> ```Makefile
> PKG_CONFIG?=pkg-config
> 
> # tieredmemdb uses the memkind allocator.
> ```

TieredMemDB


<!-- Sent from Reviewable.io -->

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @bratpiorka, @maciekpac, and @PatKamin)

___
*[`Makefile` line 13 at r3](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aHwHM-KcJd1sYD_z4:-N0aRRYs4wAMOt0-Htob:b5kxyt2) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/7f8714ff9ee1611cbc7b5dab287ee1cf4b3c19b5/Makefile#L13)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

tmdb?
</blockquote></details>

Teenage Mutant Ninj... DooBies?
Done.
___
*[`src/Makefile` line 52 at r2](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aHZ2hD-fnWb0qUgUA:-N0aRaQm-1ne1AARnsun:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/0e8e96c1981a7e015941b0308afbdee162767ecf/src/Makefile#L52)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

TieredMemDB
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

PatKamin reviewed (changes requested):
> Reviewed 3 of 3 files at r4.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 1 of 3 files reviewed, 3 unresolved discussions (waiting on @bratpiorka, @kilobyte, @maciekpac, and @PatKamin)

___
*[`src/Makefile` line 52 at r4](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aS2eC3Wgq0bBaICl0:-N0aS2eC3Wgq0bBaICl1:bvpr0c5) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/d9a9c1a34bb463276d38a8d80b76449f7c6b55a3/src/Makefile#L52)):*
> ```Makefile
> PKG_CONFIG?=pkg-config
> 
> # TieredNemDB uses the memkind allocator.
> ```

N -> M



_Code quote:_
```Makefile
TieredNemDB
```


<!-- Sent from Reviewable.io -->

PatKamin reviewed (commented):
> Reviewed 1 of 2 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 2 unresolved discussions (waiting on @bratpiorka, @kilobyte, and @maciekpac)



<!-- Sent from Reviewable.io -->

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka, @maciekpac, and @PatKamin)

___
*[`src/Makefile` line 52 at r4](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aS2eC3Wgq0bBaICl0:-N0aT7IUA-bbXAFK3iw3:bb71j7r) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/d9a9c1a34bb463276d38a8d80b76449f7c6b55a3/src/Makefile#L52)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

N -> M
</blockquote></details>

Oif.


<!-- Sent from Reviewable.io -->

maciekpac reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[`src/Makefile` line 52 at r4](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aS2eC3Wgq0bBaICl0:-N0cAMhPDIUCUaImXCRR:b32wyo) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/d9a9c1a34bb463276d38a8d80b76449f7c6b55a3/src/Makefile#L52)):*
<details><summary><i>Previously, kilobyte (Adam Borowski) wrote…</i></summary><blockquote>

Oif.
</blockquote></details>

up


<!-- Sent from Reviewable.io -->

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[`src/Makefile` line 52 at r4](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aS2eC3Wgq0bBaICl0:-N0cAfq8-Zmk5imJpsqR:boaamax) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/d9a9c1a34bb463276d38a8d80b76449f7c6b55a3/src/Makefile#L52)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

up
</blockquote></details>

Did I miss this somewhere else, or...?


<!-- Sent from Reviewable.io -->

maciekpac reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[`src/Makefile` line 52 at r4](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0aS2eC3Wgq0bBaICl0:-N0cAmzx3vWI9a-OqZH2:b1pmqie) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/d9a9c1a34bb463276d38a8d80b76449f7c6b55a3/src/Makefile#L52)):*
<details><summary><i>Previously, kilobyte (Adam Borowski) wrote…</i></summary><blockquote>

Did I miss this somewhere else, or...?
</blockquote></details>

I'm sorry, looked at wrong revision


<!-- Sent from Reviewable.io -->

PatKamin reviewed (approved):
> Reviewed 1 of 3 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 1 unresolved discussion (waiting on @bratpiorka)



<!-- Sent from Reviewable.io -->

bratpiorka reviewed (changes requested):
> Reviewed 1 of 2 files at r2, 1 of 2 files at r3, 3 of 3 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 1 unresolved discussion (waiting on @kilobyte)

___
*[`-- commits` line 2 at r9](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0e-s1S4NWsOCK6m2xX):*
Set the default allocator to memkind


<!-- Sent from Reviewable.io -->

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 1 unresolved discussion (waiting on @bratpiorka)

___
*[`-- commits` line 2 at r9](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N0e-s1S4NWsOCK6m2xX):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

Set the default allocator to memkind
</blockquote></details>

Is there a reason to use an additional transitive verb instead of a direct action?  I thought the latter is preferred in commit messages.


<!-- Sent from Reviewable.io -->

bratpiorka reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kilobyte)



<!-- Sent from Reviewable.io -->

kilobyte committed:
Replace README with TMDB-specific description.

PatKamin reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: 3 of 4 files reviewed, 9 unresolved discussions (waiting on @bratpiorka, @kilobyte, @maciekpac, and @PatKamin)

___
*[`README.md` line 5 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hDRAb98SFz1Fss--4:-N1hDRAb98SFz1Fss--5:bgm4p61) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L5)):*
> ```Markdown
> 
> This project is a fork of [Redis](https://redis.io), adapted for systems
> with multiple memory tiers.  New memory technologies include memories that
> ```

Double whitespace, remove one.



_Code quote:_
```Markdown
tiers.  New
```
___
*[`README.md` line 15 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hERnS-Nyc7BRuqGhM:-N1hERnS-Nyc7BRuqGhN:b-jcvh71) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L15)):*
> ```Markdown
> 
> This README doesn't describe usual usage of Redis.  We assume you already
> have general experience with Redis.
> ```

Maybe add a link for further reading on Redis?
___
*[`README.md` line 26 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hG6xz-GR9cqOv60R6:-N1hG6xz-GR9cqOv60R7:b-wzvlpx) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L26)):*
> ```Markdown
> of memory tiers, you need the following extra dependencies:
> 
>     * ndctl and daxctl
> ```

Add a minimum version.
___
*[`README.md` line 27 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hEu9b2VS_6gWniRWb:-N1hEu9b2VS_6gWniRWc:bckdwg5) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L27)):*
> ```Markdown
> 
>     * ndctl and daxctl
>     * memkind (included in tarballs)
> ```

memkind is included as a git submodule. You don't have to download and install it on your own. You suggest here that I need this extra dependency to be added manually.
___
*[`README.md` line 31 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hGRjYAIcq7g8Ps9mR:-N1hGRjYAIcq7g8Ps9mS:b-411h24) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L31)):*
> ```Markdown
> Building TieredMemDB is as simple as:
> 
>     % make
> ```

What about memkind as a submodule? Describe here both ways of building and installing TieredMemDB: by hand from sources and from a tarball.
___
*[`README.md` line 39 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hGyuM1UTHO3WEQSZ9:-N1hGyuM1UTHO3WEQSZA:b-300839) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L39)):*
> ```Markdown
> 
> The crux of TieredMemDB is the tiered allocator, which beside `MALLOC=libc`
> and `MALLOC=jemalloc` supported by Redis adds:
> ```

(...) environment variable



_Code quote:_
```Markdown
`MALLOC=jemalloc`
```
___
*[`README.md` line 43 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hHAyJ6OlCnKImDD4h:-N1hHAyJ6OlCnKImDD4i:b-pvsl3a) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L43)):*
> ```Markdown
>     % make MALLOC=memkind
> 
> but in release tarballs this is already the default.
> ```

What is the default when building from sources with memkind as a submodule?
___
*[`README.md` line 51 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hIOkq7Iu1bWKiFwsG:-N1hIOkq7Iu1bWKiFwsH:b-s6lwwc) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L51)):*
> ```Markdown
> The set of patches over base Redis that comprises TieredMemDB can be applied
> onto your modified versions of Redis as well.  To do that, or to update
> Redis to a new minor version (because of eg. a security update), you rebase
> ```

e.g. ?
___
*[`README.md` line 57 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hOZGZCwUlIQDCup16:-N1hOZGZCwUlIQDCup17:bnwvxv9) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L57)):*
> ```Markdown
>     % git diff 6.2.6 @
> 
> which can then be applied via `patch` over, in this case, Redis 6.2.6.
> ```

You might add the whole command here. Although I think that this whole section can be removed as there are possible problems with conflicts when applying changes.


<!-- Sent from Reviewable.io -->

bratpiorka reviewed (changes requested):
on my system git describe shows bad tag:

[rrudnick@localhost tmdb]$ git describe
6.2.5-44-g51cab33d7

> Reviewed 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-:-N2Addh-BnvMKpZ8M6bp:bii4jrr)* status: all files reviewed, 9 unresolved discussions (waiting on @kilobyte and @PatKamin)

___
*[`README.md` line 27 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hEu9b2VS_6gWniRWb:-N1i5xdkBgmxpwrms5aM:b-pnlr9l) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L27)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

memkind is included as a git submodule. You don't have to download and install it on your own. You suggest here that I need this extra dependency to be added manually.
</blockquote></details>

+1 I would only mention Memkind in main section and not here


<!-- Sent from Reviewable.io -->

kilobyte committed:
Don't re-bootstrap autoconfage if it's already there.

configure will try to update itself if able, silently falling back to
shipped version.  This is the historic model that autoconf has been
designed for -- when end-users were not expected to have adequate
devel environment.

kilobyte committed:
Ship files produced by autoconf.

Explicitly listing required files is a nasty hack, but memkind is in a
state of flux while autoconf is hardly a moving target these decades,
thus let's err on hard-coding the list of autoconf products.

bratpiorka reviewed (approved):
> Reviewed 3 of 3 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 9 unresolved discussions (waiting on @kilobyte and @PatKamin)



<!-- Sent from Reviewable.io -->

bratpiorka reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 11 unresolved discussions (waiting on @kilobyte and @PatKamin)

___
*[`README.md` line 27 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50HoSED3yYScFYz1Fe:-N50HoSED3yYScFYz1Ff:bfixe3p) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L27)):*
> ```Markdown
> 
>     * ndctl and daxctl
>     * memkind (included in tarballs)
> ```

Please remove memkind from dependency list at all - I got feedback that this is confusing
___
*[`README.md` line 38 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50IBaB-ETzYJiEf3dd:-N50IBaB-ETzYJiEf3de:b6oc9ne) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L38)):*
> ```Markdown
>     % make test
> 
> The crux of TieredMemDB is the tiered allocator, which beside `MALLOC=libc`
> ```

This is unexpected that someone will build TieredMemDB with allocator other than memking. IMHO this fragment should be removed to avoid confusion. 


<!-- Sent from Reviewable.io -->

bratpiorka reviewed (changes requested):
this works better for me: 
git describe --tags
6.2.6-14-ge162fbfac

> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-:-N50MgYK8XwpzcSn-Ose:b5rtt14)* status: all files reviewed, 12 unresolved discussions (waiting on @kilobyte and @PatKamin)

___
*[`Makefile` line 32 at r11](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50MqrUEFBNWiVsjFjv:-N50MqrUEFBNWiVsjFjw:bi9z6d8) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/e162fbfac55412750fe7e646748a55fbf8cbc9a1/Makefile#L32)):*
> ```Makefile
> 	test-driver
> 
> TARBALL=tmdb-$(shell git describe)
> ```

git describe --tags works better for me


<!-- Sent from Reviewable.io -->

kilobyte reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 12 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[`Makefile` line 32 at r11](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50MqrUEFBNWiVsjFjv:-N54_1pH7pMMWfnO174_:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/e162fbfac55412750fe7e646748a55fbf8cbc9a1/Makefile#L32)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

git describe --tags works better for me
</blockquote></details>

Done.
___
*[`README.md` line 5 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hDRAb98SFz1Fss--4:-N1hPb317F_I8aMxEgCn:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L5)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Double whitespace, remove one.
</blockquote></details>

Done.
___
*[`README.md` line 15 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hERnS-Nyc7BRuqGhM:-N1hPrfc7kWv-bz8az9y:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L15)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Maybe add a link for further reading on Redis?
</blockquote></details>

Done.
___
*[`README.md` line 26 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hG6xz-GR9cqOv60R6:-N54_K1Y15zNFwBO5Pu-:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L26)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Add a minimum version.
</blockquote></details>

Done.
___
*[`README.md` line 27 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hEu9b2VS_6gWniRWb:-N54_r8b66i9c24Z7S7H:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L27)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

+1 I would only mention Memkind in main section and not here
</blockquote></details>

Done.
___
*[`README.md` line 27 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50HoSED3yYScFYz1Fe:-N54_rTk0EYO5aZrR6A-:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L27)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

Please remove memkind from dependency list at all - I got feedback that this is confusing
</blockquote></details>

Done.
___
*[`README.md` line 31 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hGRjYAIcq7g8Ps9mR:-N54_tQyBwrVWlj3G_4s:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L31)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

What about memkind as a submodule? Describe here both ways of building and installing TieredMemDB: by hand from sources and from a tarball.
</blockquote></details>

Done.
___
*[`README.md` line 38 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N50IBaB-ETzYJiEf3dd:-N50TgDx-udcMcdHwG2i:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L38)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

This is unexpected that someone will build TieredMemDB with allocator other than memking. IMHO this fragment should be removed to avoid confusion. 
</blockquote></details>

Done.
___
*[`README.md` line 39 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hGyuM1UTHO3WEQSZ9:-N50TlLFCvjzYtlyV4gL:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L39)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

(...) environment variable
</blockquote></details>

Done.
___
*[`README.md` line 43 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hHAyJ6OlCnKImDD4h:-N50Tnd299_ZVnxCSFjn:b-pj3lba) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L43)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

What is the default when building from sources with memkind as a submodule?
</blockquote></details>

This is the git tree.
___
*[`README.md` line 51 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hIOkq7Iu1bWKiFwsG:-N54ZA7r7KayndMJ9Y02:b-tvfakd) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L51)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

e.g. ?
</blockquote></details>

[Wikipedia](https://en.wikipedia.org/wiki/List_of_Latin_phrases_(E)#cite_note-27) says both are valid, but it seems that `e.g.` is indeed more widespread, done.
___
*[`README.md` line 57 at r10](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8#-N1hOZGZCwUlIQDCup16:-N54Zm4j5CHI-75aIww1:b-va7b0i) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/51cab33d7504c33fe7bfa91ba3af4e39a50e422b/README.md#L57)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

You might add the whole command here. Although I think that this whole section can be removed as there are possible problems with conflicts when applying changes.
</blockquote></details>

Oh well...


<!-- Sent from Reviewable.io -->

PatKamin reviewed (approved):
> Reviewed 2 of 3 files at r7, 3 of 3 files at r12, 1 of 1 files at r13, 1 of 1 files at r14, 1 of 1 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: all files reviewed, 3 unresolved discussions (waiting on @bratpiorka)



<!-- Sent from Reviewable.io -->

bratpiorka reviewed (approved):
> Reviewed 2 of 3 files at r12, 1 of 1 files at r13, 1 of 1 files at r14, 1 of 1 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kilobyte)



<!-- Sent from Reviewable.io -->

kilobyte committed:
Build only the memkind library but no examples/tests/benchmarks.

They have additional dependencies; lacking them results in spurious
failures that are then ignored, this gives confusing messages.

bratpiorka reviewed (approved):
> Reviewed 1 of 1 files at r16, 4 of 4 files at r18, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kilobyte)



<!-- Sent from Reviewable.io -->

PatKamin reviewed (approved):
> Reviewed 3 of 4 files at r18, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kilobyte)



<!-- Sent from Reviewable.io -->
